### ISSUE TITLE
Undefined method `gravity' with MiniMagick 4.0.0.rc

### ISSUE THREAD

**Issue by lephyrius** [minimagick/minimagick#264]
I tried the new 4.0.0.rc and I'm getting the error:

```
NoMethodError: undefined method `gravity' for #<MiniMagick::Tool::Mogrify:0x007fa521c62280>
```

With this code:

``` rb
image.combine_options do |c|
  c.gravity :center
end
```

**Comment by qiuxiaoj** [comment]
```gm mogrify -resize x256 -gravity Center +profile * -crop 256x256+0+0 test.jpg```
It is support in GraphicsMagick 1.3.18, No test in latest.

**Comment by NARKOZ** [comment]
@janko-m do you know which option should I use instead of `gravity` in GraphicsMagick 1.3.21?

**Comment by janko-m** [comment]
GraphicsMagick doesn't have "-gravity" for `mogrify`, but it seems to have it for `convert`. For using `convert` you would have to give up the `MiniMagick::Image` interface (because it uses only `mogrify` under the hood) and use `MiniMagick::Tool::Convert` directly (which I would recommend anyways :wink:). So for example @qiuxiaoj's command could be written like this:

```rb
MiniMagick::Tool::Convert.new do |convert|
  convert << "test.jpg"
  convert.resize("x256")
  convert.gravity("Center")
  convert.profile.+("*")
  convert.crop("256x256+0+0")
  convert << "output.jpg"
end
```

**Comment by NARKOZ** [comment]
@janko-m thank you. It worked.

**Comment by felixbuenemann** [comment]
@janko-m Please re-open, graphicsmagick's mogrify does support `-gravity`, it's just not listed in `-help`. It's documented on the GM homepage: http://www.graphicsmagick.org/GraphicsMagick.html#details-gravity

**Comment by janko-m** [comment]
I think it would be much better and simpler if you filed an issue with GraphicsMagick to add the option to the `-help` output. If an option for a CLI isn't listed in its help page, it effectively doesn't exist.

**Comment by felixbuenemann** [comment]
@janko-m I'm planning to do that, but it usually takes a few years for new versions to trickle down into server lts versions, so I don't think that's sufficient.

**Comment by janko-m** [comment]
Wow, few years for a release sounds reaaally slow, at especially compared to ImageMagick's fast release cycle. You're right then, I pushed the new version (4.2.10) with the appropriate fix.

**Comment by felixbuenemann** [comment]
@janko-m Thanks, I found [graphicsmagick bug 302](http://sourceforge.net/p/graphicsmagick/bugs/302/) which lists the missing options. I'll come up with a PR to add them in the same way as `-gravity`. I tried them with a graphicsmagick version from 2010 and all were working, so they must've been around for a while. It probably makes sense to use a Set for the cli_options so no options are added twice, once a fixed graphicsmagick release comes out (it's fixed in the gm mercurial repo since 2015-06-14).

**Comment by felixbuenemann** [comment]
@janko-m Please take a look at PR #323 which should add all the missing gm mogrify options, but only if they are actually missing.

**Comment by dongtong** [comment]
brew install imagemagick graphicsmagick libpng jpeg.
 It will works

**Comment by felixbuenemann** [comment]
@dongtong This has been fixed in v4.2.10. There's no need to install imagemagick when graphicsmagick should be used.
